src/sidebar/services/frame-sync.js

```diff
@@ -181,19 +181,33 @@ export class FrameSyncService {
   _connectGuest(port) {
     /** @type {PortRPC<GuestToSidebarEvent, SidebarToGuestEvent>} */
     const guestRPC = new PortRPC();
+    this._guestRPC.push(guestRPC);
+
+    /** @type {string|null} */
+    let frameIdentifier;
 
     // Update document metadata for this guest. We currently assume that the
     // guest will make this call once after it connects. To handle updates
     // to the document, we'll need to change `connectFrame` to update rather than
     // add to the frame list.
     guestRPC.on('documentInfoChanged', info => {
+      frameIdentifier = info.frameIdentifier;
       this._store.connectFrame({
         id: info.frameIdentifier,
         metadata: info.metadata,
         uri: info.uri,
       });
     });
 
+    guestRPC.on('frameDestroyed', () => {
```

Comment:
Could `frameDestroyed` RPC event be received before the `documentInfoChanged`? If so, it will fail to destroy the frame.

Reply:
After `frameDestroyed` is received, no more RPC messages will be received from the guest. If we receive `frameDestroyed` first, then we will not receive `documentInfoChanged` and the frame will never be added to the store.

Reply:
It is true, good point.